_src/custom-date-format.md

```diff
@@ -6,22 +6,22 @@ attribute is used to provide the logic for handling the custom representation.
 
 !PLAYGROUND 7185eb211a4822ce97184ae25fedda91
 ```rust
-use chrono::{DateTime, Utc};
+use chrono::NaiveDateTime;
 use serde::{Serialize, Deserialize};
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct StructWithCustomDate {
     // DateTime supports Serde out of the box, but uses RFC3339 format. Provide
     // some custom logic to make it use our desired format.
     #[serde(with = "my_date_format")]
-    pub timestamp: DateTime<Utc>,
+    pub timestamp: NaiveDateTime,
```

Comment:
Is NaiveDateTime the right type here? Chrono documentation seems to feature DateTime more prominently (from skimming the crate root). There is a note about NaiveDateTime saying it _"can be quite low-level"_ and _"mostly useful for building blocks for higher-level types"_.

Reply:
You're totally right, DateTime should always be preferred over NaiveDateTime. I initially chose to change it in order to keep the example as close to the original as possible, but without the usage of the deprecated function. I updated it to return a DateTime<Utc> again, but that required me to change the format of the Timestamp. Is it okay now?

Reply:
Would there be any way to keep DateTime\<Utc\> in the data structure as before, and also keep the original string format as before, without using any deprecated function?

Reply:
Will look at it and update the PR if I got it working

Reply:
The chrono docs are a bit hard to grasp, but as I understand it, we would need to first create a NaiveDateTime, and then convert that into a DateTime<Utc> afterwards. Reason is that the shorter way via DateTime::parse_from_str is not possible, since it [requires a TimeZone](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.from_naive_utc_and_offset) to be present in the input str. When doing it this way, the Code would look like this:

```rust
let s = String::deserialize(deserializer)?;
let dt = NaiveDateTime::parse_from_str(&s, FORMAT).map_err(serde::de::Error::custom)?;
Ok(DateTime::<Utc>::from_naive_utc_and_offset(dt, Utc))
```

This would still use the original string format, but without using any deprecated functions. Would it be fine this way, @dtolnay?

Reply:
Good find. It looks like you are on the right track according to the docs. _"This is a low-level method, intended for use cases such as deserializing a DateTime"_

Reply:
Thank you. I updated the PR accordingly.